### ISSUE TITLE
Add POSIX warning for ${@%...} and similar

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1236]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
echo "${@%/*}"

```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

A [POSIX warning](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02):

>The following four varieties of parameter expansion provide for substring processing. [...]. If parameter is '#', '*', or '@', the result of the expansion is unspecified.
